File: Linked List/3063. Linked List Frequency/3063.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def frequenciesOfElements(self, head: Optional[ListNode]) -> Optional[ListNode]:
        freq = defaultdict(int)
        curr = head
        while curr:
            freq[curr.val] += 1
            curr = curr.next
        
        dummy = ListNode()
        curr = dummy
        for i,v in freq.items():
            new_node = ListNode(v, None)
            curr.next = new_node
            curr = curr.next

        return dummy.next

File: Linked List/3063. Linked List Frequency/3063.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* frequenciesOfElements(ListNode* head) {
        map<int,int> freq;
        ListNode* curr = head;
        while (curr) {
            freq[curr->val]++;
            curr = curr->next;
        }
        ListNode* dummy = new ListNode();
        curr = dummy;
        for (auto itr : freq) {
            ListNode* newNode = new ListNode(itr.second, nullptr);
            curr->next = newNode;
            curr = curr->next;
        }
        return dummy->next;
    }
};